### PR TITLE
allow numpy-like boolean-list indexing in pytorch

### PR DESCRIPTION
**Pull request by ruochunz**
Suggested fix to issue #6773, the fix allows numpy-like boolean-list indexing in pytorch

### CODE REVIEW

**Comment by soumith** [comment]
@ruochunz could you please add a unit test for this case, so that we dont miss it in any future refactors

**Comment by ruochunz** [comment]
Hi @soumith , something like this?

**Comment by ruochunz** [comment]
Yes, I agree. `legacy_new_from_data` is not used anywhere else, so it should be renamed to make it explicit that it is specific to indexing.

**Review by colesbury — APPROVED** [review]
Thanks for fixing this!

File: torch/csrc/utils/tensor_new.cpp [hunk 1]
```diff
@@ -449,11 +449,19 @@ Tensor legacy_tensor_new(const Type& type, PyObject* args, PyObject* kwargs) {
...
     c10::optional<Device> device,
     PyObject* data) {
-  return internal_new_from_data(type, device, data, false, false, false);
+  // Specific to tensor indexing, converts an indexing list to an 
+  // indexing tensor (type Byte or Long)
+  ScalarType scalar_type = infer_scalar_type(data);
+  if (scalar_type == ScalarType::Byte){
+      auto& idx_type = type.toScalarType(scalar_type);
+      return internal_new_from_data(idx_type, device, data, false, false, false);
+  }else{
```
[the remark this answers is not in the document]
Comment (ruochunz) [reply torch/csrc/utils/tensor_new.cpp]: Have just added spaces where required in commit 4aaca56

File: torch/csrc/utils/tensor_new.cpp [hunk 2]
```diff
@@ -449,11 +449,19 @@ Tensor legacy_tensor_new(const Type& type, PyObject* args, PyObject* kwargs) {
   throw std::runtime_error("new(): invalid arguments");
 }
 
-Tensor legacy_new_from_data(
+Tensor index_tensor_from_data(
```
[the remark this answers is not in the document]
Comment (ruochunz) [reply torch/csrc/utils/tensor_new.cpp]: Sure, have just renamed it in commit 4aaca56

**Pull request closed without merging**